src/zenml/models/code_repository_models.py

```diff
@@ -0,0 +1,128 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Models representing code repositories."""
+
+from typing import Any, Dict, Optional, Union
+from uuid import UUID
+
+from pydantic import BaseModel, Field
+
+from zenml.config.source import Source
+from zenml.models.base_models import (
+    BaseRequestModel,
+    BaseResponseModel,
+    WorkspaceScopedRequestModel,
+    WorkspaceScopedResponseModel,
+    update_model,
+)
+from zenml.models.constants import STR_FIELD_MAX_LENGTH
+from zenml.models.filter_models import WorkspaceScopedFilterModel
+
+# ---- #
+# BASE #
+# ---- #
+
+
+class CodeRepositoryBaseModel(BaseModel):
+    """Base model for code repositories."""
+
+    name: str = Field(
+        title="The name of the code repository.",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    config: Dict[str, Any] = Field(
+        description="Configuration for the code repository."
+    )
+    source: Source = Field(description="The code repository source.")
```

Comment:
Q: so we're not actually storing the code itself, just the metadata / hash etc reference necessary for us to be able to retrieve that code later on? Was just asking as we had to update one of our column sizes recently and if we were storing raw code then potentially we'd need to make that nice and large, too.

Reply:
This attribute just points to the source (e.g. `zenml.integrations.github.code_repositories.GitHubCodeRepository`) of the class that implements the code repository abstraction. To answer the question: Correct, we only store credentials to access the code later on, no code is stored on our side.